vendor/knative.dev/pkg/test/spoof/openshift_error.go

```diff
@@ -0,0 +1,44 @@
+package spoof
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+// OpenShiftErrorRetryChecker implements the ErrorRetryChecker for retrying on OpenShift error.
+func OpenShiftErrorRetryChecker(err error) (bool, error) {
+	if isUnknownAuthority(err) {
+		return true, fmt.Errorf("retrying for certificate signed by unknown authority: %w", err)
+	}
+	return false, err
+}
+
+// OpenShiftResponseRetryChecker implements the ResponseChecker for retrying on OpenShift response error.
+// - 404 until the route is propagated to the proxy
+// - 503 to account for Openshift route inconsistency (https://jira.coreos.com/browse/SRVKS-157)
+func OpenShiftResponseRetryChecker(resp *Response) (bool, error) {
```

Comment:
I think we should give this the same behavior as today, i.e. make it pass only after it has seen 5 successes in a row.

Reply:
Actually can't do that, I think :cry:.